Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/PerformanceGallery/PerformanceGallery.cs

```diff
@@ -0,0 +1,236 @@
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+using System.IO;
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(Xamarin.Forms.Core.UITests.UITestCategories.Performance)]
+#endif
+
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.None, 0, "Performance Testing")]
+	public class PerformanceGallery : TestContentPage
+	{
+		const string Fail = "FAIL";
+		const string Next = "Next Scenario";
+		const string NextButtonId = "NextButton";
+		const string TestRunRefId = "TestRunRefId";
+		const string Pending = "PENDING";
+		const string Success = "SUCCESS";
+		const double Threshold = 0.25;
+
+		string _DeviceIdentifier = "";
+		string _DeviceIdiom;
+		string _DeviceModel;
+		string _DevicePlatform;
+		string _DeviceVersionNumber;
+		string _BuildInfo;
+		PerformanceProvider _PerformanceProvider = new PerformanceProvider();
+		PerformanceTracker _PerformanceTracker = new PerformanceTracker();
+		List<PerformanceScenario> _TestCases = new List<PerformanceScenario>();
+		int _TestNumber = 0;
+
+		PerformanceViewModel ViewModel => BindingContext as PerformanceViewModel;
+
+		protected override async void Init()
+		{
+			_BuildInfo = GetBuildNumber();
+
+			_DeviceIdentifier = CrossDeviceInfo.Current.Id;
+			_DeviceIdiom = CrossDeviceInfo.Current.Idiom.ToString();
+			_DeviceModel = CrossDeviceInfo.Current.Model;
+#if __ANDROID__ && TEST_EXPERIMENTAL_RENDERERS
+			_DevicePlatform = "Android Fast Renderers";
+#else
+			_DevicePlatform = CrossDeviceInfo.Current.Platform.ToString();
+#endif
+			_DeviceVersionNumber = CrossDeviceInfo.Current.VersionNumber.ToString();
+
+			MessagingCenter.Subscribe<PerformanceTracker>(this, PerformanceTracker.RenderCompleteMessage, HandleRenderComplete);
+
+			BindingContext = new PerformanceViewModel(_PerformanceProvider);
+			Performance.SetProvider(_PerformanceProvider);
+
+			_TestCases.AddRange(InflatePerformanceScenarios());
+
+			var nextButton = new Button { Text = Pending, IsEnabled = false, AutomationId = NextButtonId };
+			nextButton.Clicked += NextButton_Clicked;
+
+			ViewModel.TestRunReferenceId = Guid.NewGuid();
+
+			var testRunRef = new Label
+			{
+				AutomationId = TestRunRefId,
+				Text = ViewModel.TestRunReferenceId.ToString(),
+				FontSize = 6,
+				HorizontalTextAlignment = TextAlignment.Center,
+				VerticalTextAlignment = TextAlignment.Center,
+				BackgroundColor = Color.Accent,
+				TextColor = Color.White
+			};
+
+			Content = new StackLayout { Children = { testRunRef, nextButton, _PerformanceTracker } };
+
+			ViewModel.BenchmarkResults = await PerformanceDataManager.GetScenarioResults(_DeviceIdentifier);
+
+			nextButton.IsEnabled = true;
+			nextButton.Text = Next;
+		}
+
+		private static string GetBuildNumber()
+		{
+			try
+			{
+				var assembly = typeof(PerformanceGallery).GetTypeInfo().Assembly;
+				var txt = "Xamarin.Forms.Controls.BuildNumber.txt";
```

Comment:
Embedded resource that I inject with the build number from the build server. This lets me categorize the scenario results on the server side.